src/challenge-2/challenge-2.c

```diff
@@ -34,7 +31,7 @@ int main(void)
     const char** peers = node_ids(init_msg);
     const size_t num_peers = node_ids_count(init_msg);
```

Comment:
QQ: Why is `node_ids_count` returning a `size_t`, but here we make `num_peers` a `const size_t` (versus original approach of making `node_ids_count` return a `const size_t`?